linesearch.go

```diff
@@ -71,29 +75,30 @@ func (l *Linesearch) Iterate(loc *Location, xNext []float64) (EvaluationType, It
 	}
 	if l.Method.Finished(linesearchLocation) {
 		if l.lastEvalType == FunctionEval {
-			l.finished = true
```

Comment:
I think the previous check was wrong. This should be 
````
if l.lastEvalType == FunctionEval && loc.Gradient != nil{
}
````
Right? If it's a gradient-free function then we don't need to evaluate the gradient.